CollectNewspaperKarel.java

```diff
@@ -1,17 +1,50 @@
-/*
- * File: CollectNewspaperKarel.java
- * --------------------------------
- * At present, the CollectNewspaperKarel subclass does nothing.
- * Your job in the assignment is to add the necessary code to
- * instruct Karel to walk to the door of its house, pick up the
- * newspaper (represented by a beeper, of course), and then return
- * to its initial position in the upper left corner of the house.
- */
-
-import stanford.karel.*;
-
-public class CollectNewspaperKarel extends SuperKarel {
-
-	// You fill in this part
-
-}
+/*
+ * File: CollectNewspaperKarel.java
+ * --------------------------------
+ * At present, the CollectNewspaperKarel subclass does nothing.
+ * Your job in the assignment is to add the necessary code to
+ * instruct Karel to walk to the door of its house, pick up the
+ * newspaper (represented by a beeper, of course), and then return
+ * to its initial position in the upper left corner of the house.
+ */
+
+import stanford.karel.*;
+
+public class CollectNewspaperKarel extends SuperKarel {
+
+	// Karel needs to move towards the newspaper pick it up and return to its position
+	public void run() {
+		 moveToBeeper();
+		 pickTheBeeper();
+		 goBack();
+		 
+		 
+		}
+
+	private void goBack() {
+		// TODO Auto-generated method stub
+		turnAround();
+		move();
+		move();
+		move();
+		turnRight();
+		move();
+		turnRight();
+	}
+
+	private void moveToBeeper() {
+		// TODO Auto-generated method stub
+		move();
+		turnRight();
+		move();
+		turnLeft();
+		move();
+		move();
+	}
+	private void pickTheBeeper(){

```

Comment:
This method is unnecessary since it is just one line. Instead of calling this method you could have used pickBeeper() directly.